mantle/platform/api/azure/instance.go

```diff
@@ -37,10 +39,31 @@ type Machine struct {
 }
 
 func (a *API) getVMParameters(name, userdata, sshkey, storageAccountURI string, ip *network.PublicIPAddress, nic *network.Interface) compute.VirtualMachine {
+
+	// Azure requires that either a username/password be set or an SSH key.
+	//
+	//    Message="Authentication using either SSH or by user name and
+	//             password must be enabled in Linux profile."
+	//
+	// Since we don't ship their agent setting a username and password here
+	// is harmless. We set the username and password because some tests explicitly
+	// don't want to pass an SSH key via the API and this allows us to do that.
+	//
+	// The password requirements are:
+	//    Message="The supplied password must be between 6-72 characters long
+	//             and must satisfy at least 3 of password complexity requirements
+	//             from the following: Contains an uppercase character, Contains a
+	//             lowercase character, Contains a numeric digit, Contains a special
+	//             character) Control characters are not allowed"
+	password := fmt.Sprintf("%s%s%s", "ABC&", strconv.Itoa(rand.Int()), "xyz")
```

Comment:
This is not cryptographically secure.  It's better than nothing, but let's use `crypto/rand` instead.

Reply:
considering this password won't ever get used (even if something picked it up and applied it to the system we still disable password authentication for SSH) I was going for something a little more simple.

We also have to mind the password requirements, which is why I just limited it to the string representation of an integeger. I guess we can [use crypto/rand to get an Int](https://stackoverflow.com/a/32350135) instead here. Or maybe you have another more concrete suggestion.

Either way do you think the added complexity is really worth it?

Reply:
This slipped today. I promise to address it before we start running Azure tests in our pipeline.

Reply:
https://github.com/coreos/coreos-assembler/pull/2783